os/fs/smartfs/smartfs_utils.c

```diff
@@ -1342,62 +1309,32 @@ int smartfs_deleteentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *en
 		goto errout;
 	}
 
-	/* Mark this entry as inactive */
-
-	direntry = (struct smartfs_entry_header_s *)&fs->fs_rwbuffer[entry->doffset];
-#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
-#ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
-	smartfs_wrle16(&direntry->flags, smartfs_rdle16(&direntry->flags) & ~SMARTFS_DIRENT_ACTIVE);
-#else
-	direntry->flags &= ~SMARTFS_DIRENT_ACTIVE;
-#endif
-#else							/* CONFIG_SMARTFS_ERASEDSTATE == 0xFF */
-#ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
-	smartfs_wrle16(&direntry->flags, smartfs_rdle16(&direntry->flags) | SMARTFS_DIRENT_ACTIVE);
-#else
-	direntry->flags |= SMARTFS_DIRENT_ACTIVE;
-#endif
-#endif							/* CONFIG_SMARTFS_ERASEDSTATE == 0xFF */
-
-	/* Write the updated flags back to the sector */
-
-	readwrite.offset = entry->doffset;
-	readwrite.count = sizeof(uint16_t);
-	readwrite.buffer = (uint8_t *)&direntry->flags;
-	ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&readwrite);
-	if (ret < 0) {
-		fdbg("Error marking entry inactive at sector %d\n", entry->dsector);
-		goto errout;
-	}
-
-	/* Test if any entries in this sector are being used */
 
 	if ((entry->dsector != fs->fs_rootsector) && (entry->dsector != entry->dfirst)) {
-		/* Scan the sector and count used entries */
-
 		count = 0;
+		/* Scan the sector and count used entries */
 		offset = sizeof(struct smartfs_chain_header_s);
 		entrysize = sizeof(struct smartfs_entry_header_s) + fs->fs_llformat.namesize;
 		while (offset + entrysize < fs->fs_llformat.availbytes) {
 			/* Test the next entry */
-
 			direntry = (struct smartfs_entry_header_s *)&fs->fs_rwbuffer[offset];
 			if (ENTRY_VALID(direntry)) {
 				/* Count this entry */
 				count++;
 			}
 
 			/* Advance to next entry */
-
 			offset += entrysize;
 		}
 
-		/* Test if the count it zero.  If it is, then we will release the sector */
-
-		if (count == 0) {
+		/* Test if the count is one.  If it is, then we will release the sector */
+		if (count == 1) {
```

Comment:
What if we don't unchain a sector if this is the last active entry that we are deleting?
Even if it has no valid entries, it can be used for allocation of future entries with smartfs_find_availableentry().

Reply:
We're not sure next operation, if we remain it as active sector but consist of inactive entries, it will consume sectors.
I think it is not good approach, because our target is small nor flash.
If we handle big enough storage & we spend much of time for experiment regarding behavior of file operation, then your suggestion will be nice.